packages/core/src/types.ts

```diff
@@ -674,6 +674,14 @@ export interface MachineConfig<
    * The initial context (extended state)
    */
   context?: TContext | (() => TContext);
+  /**
+   * Describes the events that this machine accepts.
+   *
+   * **NOTE:** This is TypeScript-only.;
+   */
+  schema?: {
```

Comment:
wasn't `states` also allowed here before?

Reply:
Yes, but because of the overloads for the `Machine()` function, it didn't make a difference - the `TStateSchema` was still being resolved (inferred?) as `any` and the only way to "fix" that was to change this:

```diff
export function Machine<
  TContext = DefaultContext,
- TStateSchema extends StateSchema = any,
+ TStateSchema extends StateSchema = unknown,
  TEvent extends EventObject = AnyEventObject
>(
  config: MachineConfig<TContext, TStateSchema, TEvent>,
  options?: Partial<MachineOptions<TContext, TEvent>>,
  initialContext?: TContext
): StateMachine<TContext, TStateSchema, TEvent>;
```

...which caused a lot of other TypeScript issues.

I want to move away from `TStateSchema` in general as well (bigger topic), as I feel like this is something that should be either handled by better TS inference of the config object, or by something like `xstate-codegen` 🤔